Mathlib/Algebra/Group/WithOne/Basic.lean

```diff
@@ -119,13 +119,13 @@ theorem map_coe (f : α →ₙ* β) (a : α) : map f (a : WithOne α) = f a :=
 @[to_additive (attr := simp)]
 theorem map_id : map (MulHom.id α) = MonoidHom.id (WithOne α) := by
   ext x
-  induction x using WithOne.cases_on <;> rfl
+  induction x <;> rfl
```

Comment:
We have `WithOne.recOneCoe` and `WithZero.recZeroCoe`. Can we just remove `WithOne.cases_on` and `WithZero.cases_on`?